src/solvers.jl

```diff
@@ -872,37 +844,138 @@ function conicdata(m::Model)
+            constr_dual_map[d] = vec(collect(c))
+        end
+        ub = m.colUpper[idx]
+        if ub != Inf && ub != 0
+            c += 1
+            d += 1
+            push!(I, c)
+            push!(J, idx)
+            push!(V, 1.0)
+            b[c] = ub
+            push!(nonneg_rows, c)
+            constr_dual_map[d] = vec(collect(c))
+        end
+    end
+
+    if !isempty(nonneg_rows)
+        push!(con_cones, (:NonNeg,nonneg_rows))
+    end
+    if !isempty(nonpos_rows)
+        push!(con_cones, (:NonPos,nonpos_rows))
+    end
+    if !isempty(eq_rows)
+        push!(con_cones, (:Zero,eq_rows))
+    end
+
+    c, d
+end
+
+function conicdata(m::Model)
+    var_cones = Any[cone for cone in m.varCones]
+    con_cones = Any[]
+    nnz = 0
+
+    numSDPRows, numSymRows, nnz = getSDPRowsInfo(m)
+
+    linconstr = m.linconstr::Vector{LinearConstraint}
+    numLinRows = length(linconstr)
+
+    numBounds = var_range2cone!(var_cones, m)
+
+    nnz += numBounds
+    for c in 1:numLinRows
+        nnz += length(linconstr[c].terms.coeffs)
+    end
+
+    numSOCRows = getNumSOCRows(m)
+    numNormRows = length(m.socconstr)
+
+    numRows = numLinRows + numBounds + numSOCRows + numSDPRows + numSymRows
+
+    # should maintain the order of constraints in the above form
+    # throughout the code c is the conic constraint index
+    constr_dual_map = Array{Vector{Int}}(numLinRows + numBounds + numNormRows + 2*length(m.sdpconstr))
+
+    b = Array{Float64}(numRows)
+
+    I = Int[]
+    J = Int[]
+    V = Float64[]
+    sizehint!(I, nnz)
+    sizehint!(J, nnz)
+    sizehint!(V, nnz)
+
+    # Fill it up
+    tmprow = IndexedVector(Float64, m.numCols)
+
+    c, d = fillConstr!(I, J, V, b, con_cones, tmprow, constr_dual_map, 0, 0, m.linconstr, m)
+
+    @assert c == numLinRows
+    @assert d == numLinRows
+
+    c, d = fillBoundsConstr!(I, J, V, b, con_cones, constr_dual_map, c, d, m)
+
+    @assert c == numLinRows + numBounds
+    @assert d == numLinRows + numBounds
+
+    c, d = fillConstr!(I, J, V, b, con_cones, tmprow, constr_dual_map, c, d, m.socconstr, m)
+
+    @assert c == numLinRows + numBounds + numSOCRows
+    @assert d == numLinRows + numBounds + numNormRows
+
+    c, d = fillConstr!(I, J, V, b, con_cones, tmprow, constr_dual_map, c, d, m.sdpconstr, m)
+
+    if c < length(b)
+        # This happens for example when symmetry constraints are dropped with SDP
+        resize!(b, c)
+    end
+
+    m.constrDualMap = constr_dual_map
+
     f = prepAffObjective(m)
 
     # The conic MPB interface defines conic problems as
     # always being minimization problems, so flip if needed
     m.objSense == :Max && scale!(f, -1.0)
 
-    # Objective coefficients and columns of A matrix are
-    # rescaled for SDP variables
-    offdiagvars = offdiagsdpvars(m)
-    f[offdiagvars] /= sqrt(2)
-    for k in 1:length(J)
-        if offdiagvars[J[k]]
-            V[k] /= sqrt(2)
-        end
-    end
+    rescaleSDcols!(f, J, V, m)
```

Comment:
Is this the first step in your master plan to getting rid of the ``sqrt(2)`` scaling? :)